tensordict/tensordict.py

```diff
@@ -1424,6 +1424,36 @@ def reshape(
             batch_size = shape
         return TensorDict(d, batch_size, device=self.device)
 
+    def split(
+        self, split_size: Union[int, List[int]], dim: int = 0
+    ) -> List[TensorDictBase]:
+        """Splits each tensor in the TensorDict with the specified size in the given dimension, like `torch.split`.
+        Returns a list of TensorDict with the view of split chunks of tensor.
+
+        The list of TensorDict maintains the original order of the tensor chunks.
+
+        Args:
+            split_size (int) or (list(int)): size of a single chunk or list of sizes for each chunk
+            dim (int): dimension along which to split the tensor
+
+        Returns:
+            A list of TensorDict with specified size in given dimension.
+
+        """
+        dictionaries = []
+        batch_size = []
+        for key, item in self.items():
+            split_tensors = torch.split(item, split_size, dim)
+            if not dictionaries:
+                dictionaries = [dict() for _ in range(len(split_tensors))]
+                batch_size = [split_tensor.shape[:-1] for split_tensor in split_tensors]
```

Comment:
I'm a bit confused by the batch size

If you have this case

```
t = TensorDict({}, [3, 10, 5])
t.split([3, 5, 2], 1)
```
you would get 3 (empty) tensordicts of batch size `[3, 3, 5]`, `[3, 5, 5]` and`[3, 2, 5]`.
here that would not be the result i think

Reply:
The `batch_size` and the first dimensions of each of the tensors must be compliant. I previously did not consider this as legal inputs. I'll add the edge cases to the split method. On a side note, can you share more context in how empty tensordict with batch size will be used? Thanks!

Reply:
I would not compute the batch size of the tensordicts based on the size of the tensors
`TensorDict({}, [])` is not splittable
`TensorDict({"a": torch.randn(3)}, [])` is not splittable either (because you can only split along the batch size)
`TensorDict({}, [3, 4])` is splittable (an empty tensordict is like saying "everything you put in this tensordict will have a shape of `[3, 4, *other]`

Reply:
The support to empty TensorDict is now implemented. Thanks for the feedback!